CONTRIBUTING.md

```diff
@@ -2,117 +2,63 @@
+    - Or pick up a [`help wanted`](https://github.com/cosmos/ibc-go/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) issue. These issues should be a bit more involved than the good first issues and the developer working on them would benefit from some familiarity with the codebase. This type of issues may involve adding new (or extending the functionality of existing) gRPC endpoints, bumping the version of Cosmos SDK or Tendermint or fixing bugs.
+2. Participate in thoughtful discussion on that issue.
 3. If you would like to contribute:
-   1. If the issue is a proposal, ensure that the proposal has been accepted
-   2. Ensure that nobody else has already begun working on this issue. If they have,
-      make sure to contact them to collaborate
-   3. If nobody has been assigned for the issue and you would like to work on it,
-      make a comment on the issue to inform the community of your intentions
-      to begin work
-   4. Follow standard Github best practices: fork the repo, branch from the
-      HEAD of `main`, make some commits, and submit a PR to `main`
+   1. If the issue is a proposal, ensure that the proposal has been accepted.
+   2. Ensure that nobody else has already begun working on this issue. If they have, make sure to contact them to collaborate.
+   3. If nobody has been assigned for the issue and you would like to work on it, 
+   make a comment on the issue to inform the community of your intentions
+   to begin work. Then we will be able to assign the issue to you. 
+   4. Follow standard GitHub best practices: fork the repo, branch from the
+   HEAD of `main`, make some commits, and submit a PR to `main`.
       - For core developers working within the ibc-go repo, to ensure a clear
         ownership of branches, branches must be named with the convention
-        `{moniker}/{issue#}-branch-name`
-   5. Be sure to submit the PR in `Draft` mode submit your PR early, even if
-      it's incomplete as this indicates to the community you're working on
-      something and allows them to provide comments early in the development process
-   6. When the code is complete it can be marked `Ready for Review`
-   7. Be sure to include a relevant change log entry in the `Unreleased` section
-      of `CHANGELOG.md` (see file for log format)
+        `{moniker}/{issue#}-branch-name`.
+   5. Feel free to submit the PR in `Draft` mode, even if
+   the work is not complete, as this indicates to the community you're working on
+   something and allows them to provide comments early in the development process.
+   6. When the code is complete it can be marked `Ready for Review`.
+   7. Be sure to include a relevant changelog entry in the `Unreleased` section
+   of [`CHANGELOG.md`](./CHANGELOG.md) (see file for log format).
```

Comment:
Is it possible to reference a document which explains how to contribute to open source projects for newcomers? I feel like this is too much unnecessary information for regular open source contributors. I'd like to have a section where seasoned contributors can come, read for a few minutes and understand how best to contribute to our codebase. This would involve limiting the amount of information they need to read. Thoughts?